### PR TITLE
[POAE7-2865] Add multithread benchmark support for parquet reader

### PR DESCRIPTION
**Pull request by xieqi**
### What changes were proposed in this pull request?

Add multithread parquet reader benchmark

### Why are the changes needed?

Add multithread parquet reader benchmark

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Using exist UTs

### Which label does this PR belong to?

INFRA

### CODE REVIEW

File: cpp/src/compression/benchmark/ParquetBenchmark.cpp [hunk 1]
```diff
@@ -68,18 +71,99 @@ class ParquetBenchamarkReader {
...
+
+  double ratio = static_cast<double>(reader.getTotalByteSize()) /
+                 static_cast<double>(reader.getTotalCompressedSize());
+  state.counters["ratio"] = benchmark::Counter(
+      ratio, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+}
+
+void print_usage() {
+  std::cout << R"(Usage:
+    '--iter'   or '-t' to specify the number of iterations to run the benchmark
```
Comment (Ma-Jian1) [inline cpp/src/compression/benchmark/ParquetBenchmark.cpp:115]: `-i`?

File: cpp/src/compression/benchmark/ParquetBenchmark.cpp [hunk 2]
```diff
@@ -68,18 +71,99 @@ class ParquetBenchamarkReader {
...
-BENCHMARK(BM_ParquetRead);
+}  // namespace
+
+int main(int argc, char** argv) {
+  int iterations = 1;
+  int threads = 1;
+  std::string file = "lineitem.igzip";
+  std::string codec = "igzip";
+
+  for (auto i = 0; i < argc; i++) {
```
Comment (Ma-Jian1) [inline cpp/src/compression/benchmark/ParquetBenchmark.cpp:132]: small tip: you can use gflags as command line option.

Comment (xieqi) [reply cpp/src/compression/benchmark/ParquetBenchmark.cpp:132]: I think no need to introduce gflags  as dependency

**Review by Ma-Jian1 — APPROVED** [review]

File: cpp/src/compression/codec/icl_codec.h [hunk 3]
```diff
@@ -36,7 +36,8 @@ class IclCompressionCodec {
   /// \brief Create a ICL codec for the given codec name and compression level.
   ///
   /// \param[in] codec_name the name of the backend codec used by ICL, supported codec can
-  /// be "igzip", "qpl", "qat" \param[in] compression_level the compression level for the
+  /// be "igzip", "qpl", "qat"
```
Comment (yma11) [inline cpp/src/compression/codec/icl_codec.h:39]: Minor: revert this change?

**Review by winningsix — APPROVED** [review]
Let's add doc later.
